dashboard/app/helpers/aichat_openai_helper.rb

```diff
@@ -72,6 +76,60 @@ def self.get_instructions(system_prompt, level_system_prompt, retrieval_contexts
     instructions
   end
 
+  # Reports and logs usage metrics to Cloudwatch
+  def self.report_usage_metrics(usage, messages, level_id, encrypted_channel_id, user_id)
+    return unless usage
+
+    filtered = messages.reject {|message| message[:content].is_a?(String)}
+    messages_with_assets_count = filtered.count {|message| message[:content].any? {|c| c[:type] != 'text'}}
+    pdfs_count = filtered.sum {|message| message[:content].count {|c| c[:type] == 'file'}}
```

Comment:
Nit: I find these double nested blocks kind of hard to read, could we do something like:

```
pdfs_count = filtered.sum do |message|
  message[:content].count {|c| c[:type] == 'file'}
end
```